chore(deps-dev): bump vitest and @vitest/ui to 4.1.2 (coordinated)

## Context

Dependabot PR #238 updated only `vitest` to 4.1.1, leaving `@vitest/ui` at 4.0.18. `@vitest/ui` requires `vitest@"4.0.18"` as an exact peer dependency — this causes an ERESOLVE conflict in `npm ci`.

Both packages must be updated together.

Latest available: `vitest@4.1.2`, `@vitest/ui@4.1.2`

## Action

In `oia-site/package.json`, update both:
- `vitest`: `4.0.18` → `4.1.2`
- `@vitest/ui`: `4.0.18` → `4.1.2`

Run `npm install` to update the lockfile. Verify tests pass.

## Acceptance criteria

- [ ] `vitest` and `@vitest/ui` both at `4.1.2` in `package.json`
- [ ] `npm ci` succeeds
- [ ] All tests pass

--- a/oia-site/package.json
+++ b/oia-site/package.json
@@ -28,7 +28,7 @@
   },
   "devDependencies": {
     "@eslint/js": "^10.0.1",
-    "@vitest/ui": "4.0.18",
+    "@vitest/ui": "^4.1.2",
     "eslint": "^10.1.0",
     "eslint-config-prettier": "^10.1.8",
     "husky": "^9.1.7",
@@ -38,6 +38,6 @@
     "typescript": "~5.9.3",
     "typescript-eslint": "^8.57.2",
     "vite": "^8.0.3",
-    "vitest": "4.0.18"
+    "vitest": "^4.1.2"
   }
 }

